### PR TITLE
Define CRYPTO_CLOSE

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

File: draft-ietf-quic-tls.md [hunk 1]
```diff
@@ -321,8 +321,9 @@ data into the same packet.
 In general, the rules for which data can appear in packets of which
 encryption level are the same in QUIC as in TLS over TCP:
 
-u- Handshake messages MAY appear in packets of any encryption level.
-  [TODO: Alerts]
+- CRYPTO frames (Handshake messages) MAY appear in packets of any encryption level.
```
Comment (ianswett) [inline draft-ietf-quic-tls.md]: nit: I prefer without parens "CRYPTO frames containing handshake messages MAY appear..."

Is it valid to send CRYPTO frames in 0RTT?

File: draft-ietf-quic-tls.md [hunk 2]
```diff
@@ -616,16 +617,10 @@ the state of all streams, including application state bound to those streams.
...
-
-
-Errors in the TLS connection SHOULD be signaled using TLS alerts on stream 0.  A
-failure in the handshake MUST be treated as a QUIC connection error of type
-TLS_HANDSHAKE_FAILED.  Once the handshake is complete, an error in the TLS
-connection that causes a TLS alert to be sent or received MUST be treated as a
-QUIC connection error of type TLS_FATAL_ALERT_GENERATED or
-TLS_FATAL_ALERT_RECEIVED respectively.
+If TLS experiences an error, it MUST generate an appropriate alert
+as defined in {{TLS13}}; Section 6) and then provide it to QUIC,
```
Comment (chris-wood) [inline draft-ietf-quic-tls.md]: Do we worry about ossification around these alerts? Should we have a mapping from TLS->QUIC errors?

File: draft-ietf-quic-transport.md [hunk 3]
```diff
@@ -3056,6 +3059,17 @@ a real ACK.
 A EMPTY_ACK frame has no content.  That is, a EMPTY_ACK frame consists of the
 single octet that identifies the frame as a EMPTY_ACK frame.
 
+## CRYPTO_CLOSE Frame {#frame-crypto-close}
+
+The CRYPTO_CLOSE frame (type=0x20) is used to indicate connection failures
```
Comment (chris-wood) [inline draft-ietf-quic-transport.md]: Are they always failures, or do you mean alerts? CONNECTION_CLOSE can be empty to signal graceful close.

File: draft-ietf-quic-tls.md (hunk 2 above)
Comment (ekr) [reply draft-ietf-quic-tls.md]: I don't think this is going to cause ossification. This stuff is all at least obfuscated.

File: draft-ietf-quic-transport.md (hunk 3 above)
Comment (ekr) [reply draft-ietf-quic-transport.md]: They are failures.

File: draft-ietf-quic-tls.md (hunk 1 above)
Comment (ekr) [reply draft-ietf-quic-tls.md]: The problem is that this isn't restrictive. CRYPTO frames only contain HS messages.

File: draft-ietf-quic-tls.md (hunk 2 above)
Comment (chris-wood) [reply draft-ietf-quic-tls.md]: I mean clients and servers ossifying about TLS alerts as the expected error format.

**Pull request closed without merging**